trin-core/src/portalnet/overlay_service.rs

```diff
@@ -2067,9 +2068,12 @@ fn limit_enr_list_to_max_bytes(enrs: Vec<SszEnr>, max_size: usize) -> Vec<SszEnr
     }
 
     let mut total_size: usize = 0;
+
     enrs.into_iter()
         .take_while(|enr| {
-            let enr_size = enr.ssz_bytes_len();
+            // Add 4 bytes per ENR for SSZ offset overhead.
+            let enr_size = enr.ssz_bytes_len() + 4;
```

Comment:
So IIUC the offset adjustment is not accounted for in the call to `ssz_bytes_len` because the offset only appears in the encoding of a list. Would it be simpler to gradually build the list, and then call `ssz_bytes_len` on the list? That would hurt performance, but it seems simpler and less error prone.

Reply:
@jacobkaufmann If we were seeing warnings that our packets were much larger than this overhead could account for I'd say yea let's sacrifice performance to guarantee that we account for all overhead. But I reproduced the warning a lot and the highest overflow was by 20 bytes, where this change accounts for an extra ~40 bytes of overhead for 9 ENRs (observed maximum # being returned). So I feel confident that these changes are the only thing missing and will fix the issue.

Reply:
My suggestion was instead to do something like the following:

```rust
while enrs.ssz_bytes_len() > max_size {
  enrs.pop();
}
```

If `ssz_bytes_len`, when called on the `Vec`, accounts for the offset, then that is less accounting for us to do.

Reply:
Good call, that's nice.